packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditGenerators.ts

```diff
@@ -64,6 +64,12 @@ export interface FuzzTestState extends DDSFuzzTestState<SharedTreeFactory> {
 	 * and the fuzz testing model only simulates async transactions.
 	 */
 	transactionViews?: Map<ISharedTree, ITreeViewFork<typeof fuzzSchema.rootFieldSchema>>;
+	/**
+	 * NodeSchema of clients used to typecheck schema during edit generation/reducing.
+	 * If nodeSchema of a client is not present, it is set to {@link fuzzNode}, the default/initial nodeSchema is used for typechecking. This is initialized in {@link viewFromState}.
+	 * @privateRemarks TODO: The NodeSchema for each client should be properly updated if "afterSchemaChange" occurs for the client's schema when schemaOps are supported.
+	 */
```

Comment:
I think a lot of the plumbing changes in this PR would be simplier if you did

```typescript
type FuzzView = FlexTreeView<typeof fuzzSchema.rootFieldSchema> & {
	/**
	 * This client's current stored schema, which dictates allowable edits that the client may perform.
	 * @remarks - The type of this field isn't totally correct, since the supported schema for fuzz nodes changes
	 * at runtime to support different primitives (this allows fuzz testing
	 * of schema changes).
	 * However, fuzz schemas always have the same field names, so schema-dependent
	 * APIs such as the tree reading API will work correctly anyway.
	 * 
	 * TODO: The schema for each client should be properly updated if "afterSchemaChange" (or equivalent event) occurs
	 * once schema ops are supported.
	 */
	currentSchema: FuzzNodeSchema;
}
```

You'll probably need to wordwrap that better :P

Reply:
Then you'd not need to worry about plumbing state around.

You could define a similar type for the forked views as well (having a schema property on those is also good, since a transaction could edit the stored schema mid-transaction as well)

Reply:
Yeah, that makes more sense :)

It seems to not allow me to call `.fork()` on a `FuzzView` though (although it allows me to compile), so created an interface which stores the `FlexTreeView` into a `tree` field instead.

```
export interface FuzzView {
	tree: FlexTreeView<typeof fuzzSchema.rootFieldSchema>;
	/**
	 * This client's current stored schema, which dictates allowable edits that the client may perform.
	 * @remarks - The type of this field isn't totally correct, since the supported schema for fuzz nodes changes
	 * at runtime to support different primitives (this allows fuzz testing of schema changes).
	 * However, fuzz schemas always have the same field names, so schema-dependent
	 * APIs such as the tree reading API will work correctly anyway.
	 *
	 * TODO: The schema for each client should be properly updated if "afterSchemaChange" (or equivalent event) occurs
	 * once schema ops are supported.
	 */
	currentSchema: FuzzNodeSchema;
}
```

Reply:
I suggested an intersection type rather than a new object (note the `&` in the code snippet). I slightly prefer this to a new object which contains the FlexTreeView, since now we have two things called view and the naming is a bit confusing.

Reply:
updated interfaces to intersection types